packages/multichain-network-controller/src/types.ts

```diff
@@ -183,3 +211,22 @@ export type MultichainNetworkControllerMessenger = RestrictedMessenger<
   AllowedActions['type'],
   AllowedEvents['type']
 >;
+
+export const ActiveNetworksResponseStruct = object({
+  activeNetworks: array(pattern(string(), /^[^:]+:[^:]+:[^:]+$/u)),
```

Comment:
You can re-use the structs with define `@metamask/utils`:
- https://github.com/MetaMask/utils/blob/main/src/caip-types.ts#L60-L67

Using that `struct` will also type `activeNetworks` properly (the same way you did basically).